sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/OffHeapColumnVector.java

```diff
@@ -517,6 +517,7 @@ public void loadBytes(ColumnVector.Array array) {
   protected void reserveInternal(int newCapacity) {
     int oldCapacity = (this.data == 0L) ? 0 : capacity;
     if (this.resultArray != null) {
+      oldCapacity = (this.lengthData == 0L) ? 0 : capacity;
```

Comment:
Structs have a similar problem, only `nulls` is used and `data == 0`. Should we also fix these here? 

A related question, maybe we should use `nulls` instead of `data` or `length` to detect if we are resizing the column or creating a new one.

Reply:
Fair point. I'll fix it.